Consolidate D-pad CVars and clean up presets

jbodner09 opened the pull request:
* Better documents a hidden CVar that will make the D-pad on the name entry and pause screens only move a single space no matter how long the button is held, instead of mirroring the behavior of the analog stick
* Break up the Pause and File Select CVar so that the Pause screen has its own CVar for people who want to use D-pad on the file select screen but not the pause screen (closes #1094) 
* Consolidate remaining CVars into a single CVar for all text-based movement (all text boxes besides the save prompt on the pause screen, browsing shop items, and all file selection / name entry screen movement) (closes #853)

jbodner09 committed:
Consolidate D-pad CVars

jbodner09 committed:
Update other tooltip

aMannus commented:
Could you edit the enhancement presets to reflect all the new dpad options? I also noticed when some of the options were moved to the ocarina controls window, one option wasn't included in the presets. Would be nice if they could all be cleaned up.

jbodner09 commented:
Oops, I was totally going to do that and then forgot!

jbodner09 committed:
Clean up presets

jbodner09 committed:
Fix incorrect preset value

jbodner09 commented:
@aMannus I added all of the CVars from the custom control window into the presets since Argorath's change added some more in there.  I also cleaned up a couple of duplicated ones and ones that weren't in the Default setting, and added a few to the non-Default presets that seemed like they should be in there (some of which were ones I wrote to begin with, and totally should have put in there to begin with).  I did notice that there are still quite a few CVars that aren't included in the presets at all but could be, like the experimental settings besides interpolation, a few of the cheats, and a few of the develop settings.

aMannus commented:
Thank you for taking the time to clean up the presets!

As for your questions:

The enhancement presets should be a curated list of things that don't feel alien to the player when they don't set them themselves, because when they're using presets, they likely don't know what's included in them. That means that anything that doesn't fit the base game or can have other consequences should not be included in the presets. Players can still go into the enhancements themselves and change them after applying presets, but that should be up to the player. And yes, this does mean quite a few options will never be set by the presets, but I personally don't see an issue with this.

The above is the reason why I initially didn't add Dpad support for the pause menu for example. The player not knowing about holding C-up to equip items to the Dpad because they never read the option itself is likely going to lead to a lot of players being confused about what's going on, and asking in support why their Dpad items don't work.

As for the experimental ones, they're not included because they're just that, experimental. Once they lose their experimental status, we can add them to the relevant presets.

Same kind of goes for cheats as well. They're cheats, not enhancements. We shouldn't enable them through a preset without the player knowing they might be on. There's also the part where I really dislike the idea of changing settings in a completely different menu item.

So with all that, looping back to some of the changes you made:

- `CVar_SetS32("gClimbSpeed", 3);` This actually changes the difficulty of some areas, making them easier. For example, when climbing a wall with skulltulas on it. I don't think setting this above 1 should be in the presets. The push block speed one doesn't have this problem except for maybe the Poe fight in Forest Temple, hence why I think that one is fine to set it higher.
- `CVar_SetS32("gCrouchStabHammerFix", 1);` This is a large difference and many people playing OoT are used to how power crouch stabbing works. Enabling this fix in the presets doesn't sound like a good idea.
~~- `CVar_SetS32("gSkipScarecrow", 1);` This can actually bypass item logic, and is almost more like a cheat in that regard. It's planned to have this setting in the randomizer settings menu itself, but should not be in the presets.~~ I was wrong about this one, sorry.
- `CVar_SetS32("gGoronPot", 1);` Also changes the difficulty significantly, and changes something the player likely isn't aware of purely by playing the game. 

I really do appreciate you cleaning up the presets, so I hope this doesn't come across as me police-ing you around in that regard. Of course, these are all my opinions, so if there's something you disagree with or have more questions about, let me know.

briaguya0 commented:
@aMannus the skip scarecrow enhancement still requires learning scarecrow's song. I don't see how it can bypass item logic

aMannus commented:
> @aMannus the skip scarecrow enhancement still requires learning scarecrow's song. I don't see how it can bypass item logic

I thought the "still requires song" version was coming with the other enhancements in https://github.com/HarbourMasters/Shipwright/pull/1197, that's my bad then.

jbodner09 commented:
@aMannus this is a good conversation to have, and I hope more people chime in too!  You've convinced me, and I'll gladly remove the crouch stab one, but I'd really like to keep the other two you mentioned, and not just because I added the underlying features.  My guiding principle when I was adding those to the presets was what our target audience would expect from an "enhanced" version of a game they already know pretty well, because at this point, I believe our main audience is people who are already familiar with the game and are likely pretty skilled at it.  I'd love to confirm our audience demographics though.  But going with my assumption, since most people will likely be used to certain crouch behavior, you're right, we shouldn't change that core mechanic with the presets.  

Beyond just this PR, altering difficulty has never really been a consideration for me.  Instead, I want to reduce tedium.  I want to get rid of things that aren't fun.  Wasting time by climbing isn't fun.  In fact, avoiding walltulas is easy for most of our audience since they already know their behavior.  That goron pot is super not fun; it's just annoying.  Making you waste less time on it doesn't make the game any easier, but it does make it more fun.  That's actually why I thought a few of the cheats would fit well too, since the line between enhancement and cheat is already pretty blurry.  For example, drops disappearing is annoying, but rarely would affect difficulty significantly for a decently skilled player.  Not being able to use your shield with Big Goron Sword is really annoying.  Losing your Deku shield to fire is insanely annoying, although that one definitely could make the game more difficult in certain situations, so I could be convinced otherwise about it.

jbodner09 committed:
Undo crouch

aMannus commented:
I can see your point that tedium could "overrule" difficulty in a way, although that border is probably always going to be a very grey area and very subjective. I can live with the climbing speed and goron pot in that regard, and what I said before `Players can still go into the enhancements themselves and change them after applying presets` obviously goes both ways as well, so there's that.

That said, I do really believe that the options in "Cheats" are crossing that border, and shouldn't be included in the presets. Burning your shield and Biggoron's Sword being a 2 handed weapon are both deliberate design choices of the game, and should be left up to the player if they want to change them. I don't really find them solving any tedium either, they're straight up lowering the difficulty instead. The drops not dissapearing cheat I can see blurring the line, although even then, I still don't like the idea of altering options in a seperate menu purely because of UX reasons.

jbodner09 commented:
Honestly, I agree more with the UX reasons, since the whole goal of the presets was to get around our labyrinthine UI!

jbodner09 committed:
Merge branch 'develop-zhora' into consolidate-dpad

jbodner09 committed:
Merge branch 'develop-zhora' into consolidate-dpad

jbodner09 committed:
Merge branch 'develop' into consolidate-dpad
